ush/gefs_atmos_prep.sh

```diff
@@ -1,154 +1,176 @@
-#!/bin/bash
+#! /usr/bin/env bash
 
-echo "$(date -u) begin $(basename $BASH_SOURCE)"
-export PS4="${PS4}${1}: "
+source "${HOMEgfs:-${HOMEgefs}}/ush/preamble.sh"
 
-set -xa
-if [[ ${STRICT:-NO} == "YES" ]]; then
-	# Turn on strict bash error checking
-	set -eu
-fi
+#echo "$(date -u) begin $(basename $BASH_SOURCE)"
+#export PS4="${PS4}${1}: "
+
+#set -xa
+#if [[ ${STRICT:-NO} == "YES" ]]; then
+#  # Turn on strict bash error checking
+#  set -eu
+#fi
 
 export mem=$1
 export nmem=$(echo $mem|cut -c 2-)
 nmem=${nmem#0}
 
-export INIDIR=$DATA
-export OUTDIR=$GESOUT/enkf/$mem
-INITDIR=$GESOUT/init/$mem
-mkdir -p $INIDIR
-mkdir -p $OUTDIR
-mkdir -p $INITDIR
+YMD=${PDY} HH=${cyc} MEMDIR=${mem} generate_com -x COM_ATMOS_INPUT
+
+#export INIDIR=$DATA
+export OUTDIR=${COM_ATMOS_INPUT} # ${COMOUT}/$mem/atmos/INPUT
 
-cd $INIDIR
+mkdir -p $OUTDIR
+cd $DATA
+
+# Copy from set_fixed_files.sh
+#---------------------------------------------------------------------------
+# Set directory names and file names for orog data
+# The old and new (support fractional grid) orog data have different file names
+#---------------------------------------------------------------------------
+export FRAC_ORO="yes"
+if [ "${FRAC_ORO:-"no"}" = "yes" ]; then
+  if  [ ${CASE} == 'C48' ] ; then
+    OCNRES='500'
+  elif [ ${CASE} == 'C96' ] ; then
+    OCNRES='100'
+  elif [ ${CASE} == 'C192' ] ; then
+    OCNRES='050'
+  elif [ ${CASE} == 'C384' ] || [ ${CASE} == 'C768' ] || [ ${CASE} == 'C1152' ]; then
+    OCNRES='025'
+  fi
+  ORO_DIR="${CASE}.mx${OCNRES}_frac"
+  ORO_NAME="oro_${CASE}.mx${OCNRES}"
+else
+  ORO_DIR="${CASE}"
+  ORO_NAME="${CASE}_oro_data"
+fi
 
 if [[ $mem = c00 ]] ;then
-	# Control intial conditions from current GFS cycle
-	ATMFILE=$COMINgfs/gfs.t${cyc}z.atmanl.nc
-	if [[ -f $ATMFILE ]]; then
-		$NCP $ATMFILE $INIDIR
-		export ATM_FILES_INPUT="gfs.t${cyc}z.atmanl.nc"
-	else
-		msg="FATAL ERROR in $(basename $BASH_SOURCE): GFS atmospheric analysis file $ATMFILE not found!"
-		echo "$msg"
-		export err=101
-		err_chk || exit $err
-	fi
-	export CONVERT_SFC=".true."
+  # Control intial conditions from current GFS cycle
+  export ATM_FILES_INPUT="gfs.t${cyc}z.atmanl.nc"
```

Comment:
This is only one file, the variable name shouldn't be plural.
```suggestion
  export ATM_FILE_INPUT="gfs.t${cyc}z.atmanl.nc"
```

Reply:
Good insight! Thanks.

Reply:
`ATM_FILES_INPUT` is the input for `chgres_cube.sh`